CRM/Contact/Import/Parser/Contact.php

```diff
@@ -2251,4 +2249,365 @@ public static function getParameterForParser($count) {
 
   }
 
+  /**
+   * Format contact parameters.
+   *
+   * @todo this function needs re-writing & re-merging into the main function.
+   *
+   * Here be dragons.
+   *
+   * @return bool
+   */
+  protected function formatContactParameters(&$values, &$params) {
+    // Crawl through the possible classes:
```

Comment:
I've moved this over from DeprecatedUtils & fixed it for this purpose, as well as starting the long slow process of tidying it up. It should be close to removable from deprecated utils now - getting rid of that file being a goal